.github/scripts/metrics_checker.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -o pipefail
+
+# search for any "new" or modified metric emissions
+metrics_modified=$(git --no-pager diff HEAD origin/main | grep -i "SetGauge\|EmitKey\|IncrCounter\|AddSample\|MeasureSince\|UpdateFilter")
+# search for PR body or title metric references
+metrics_in_pr_body=$(echo "$PR_BODY" | grep "metric")
```

Comment:
should we use `grep -i` here to catch mixed case? eg. "Metrics for X updated to ...'"